### PR TITLE
Use arxiv's compiler

### PR DESCRIPTION
**Pull request by stefanc-ai2**
https://github.com/allenai/scholar/issues/27321

Changes:
1. Use TeXLive from Arxiv instead of from `apt-get`.
2. Change config.ini to point to where TeXLive is installed

Testing: manual testing
- can compile 2 latex files that were previously failed to compile
- can compile 1 latex file that was successfully compiled

Also, the image size balloons to 18GB. We're gonna need a bigger boat, @ca16.

### CODE REVIEW

File: data-processing/Dockerfile [hunk 1]
```diff
@@ -20,6 +17,19 @@ RUN perlbrew init
 RUN perlbrew --notest install perl-5.22.4
 RUN perlbrew install-cpanm
 
+# Install LaTeX suite
+# This download will take an extremely long time to install, so we install it first
+RUN apt-get install -y wget
+RUN wget https://arxiv-web-static1.s3.amazonaws.com/semantic_scholar_20210412/arXivTeXLive2020.tar.gz
```
Comment (stefanc-ai2) [inline data-processing/Dockerfile:23]: I also have this copied to `s3://ai2-s2-scholarphi-pipeline-prod/arxiv-compiler/` for fun/backup, but getting it from S3 bucket requires AWS creds and cli. Any idea of a simple way to get it from our S3? If not, maybe it's okay to leave it this way?

File: data-processing/requirements.txt [hunk 2]
```diff
@@ -3,7 +3,7 @@ beautifulsoup4==4.8.0
 lxml==4.4.2
 nltk==3.4.5
 numpy==1.17.4
-opencv-python==3.4.8.29
+opencv-python-headless==3.4.8.29
```
Comment (stefanc-ai2) [inline data-processing/requirements.txt:6]: Ran into missing `libSM.so` error. Getting the `headless` version is the solution according to https://stackoverflow.com/a/56738577.

File: data-processing/Dockerfile (hunk 1 above)
Comment (ca16) [reply data-processing/Dockerfile:23]: > for fun

🤣

File: data-processing/Dockerfile [hunk 3]
```diff
@@ -35,35 +45,26 @@ RUN python get-pip.py --force-reinstall
 RUN curl -sL https://deb.nodesource.com/setup_12.x | bash
 RUN apt-get install -y nodejs
 
-# Let ImageMagick to read and write PDFs
-RUN sed -i '/pattern="PDF"/s/rights="none"/rights="read | write"/' /etc/ImageMagick-6/policy.xml
```
Comment (ca16) [inline data-processing/Dockerfile:39]: Do we not need this anymore?

Comment (stefanc-ai2) [reply data-processing/Dockerfile:39]: Talked to Andrew and likely we don't need this anymore

**Review by ca16 — APPROVED** [review]
Exciting...

File: data-processing/Dockerfile (hunk 3 above)
Comment (stefanc-ai2) [reply data-processing/Dockerfile:39]: Turns out, we still need this. Latex compilation fails without this.

File: data-processing/Dockerfile (hunk 1 above)
Comment (cmwilhelm) [reply data-processing/Dockerfile:23]: Kind of gross but we could parameterize the loc with a build arg for s3 path. Use the one that's hardcoded here as a default but allow us to use our own within our land of credentials (which all boxes building this should have).

**Review by cmwilhelm — APPROVED** [review]
Fun fun fun.

Before merge I would recommend a couple things:

1. Verify that the scholarphi-pipeline image can still be built and run correctly when extending this.
2. Run the pipeline on a few test papers before/after to make sure the output is the same.

File: data-processing/Dockerfile (hunk 1 above)
Comment (stefanc-ai2) [reply data-processing/Dockerfile:23]: Ah I forgot that this is an open-sourced project. Changing it to S2's S3 will be problematic.

**Comment by stefanc-ai2** [comment]
> Fun fun fun.
> 
> Before merge I would recommend a couple things:
> 
> 1. Verify that the scholarphi-pipeline image can still be built and run correctly when extending this.
> 2. Run the pipeline on a few test papers before/after to make sure the output is the same.

1. Verified `scholarphi-pipeline` image can still be built and run
2. Compiled latex files from `2009.12303v4` and `2012.09607v2` using old and new latex compilers and the resulting pdfs are identical.

**Pull request merged**